src/shared/mount-util.c

```diff
@@ -772,6 +774,29 @@ int bind_mount_in_namespace(
         assert(src);
         assert(dest);
 
+        r = namespace_open(target, NULL, &mntns_fd, NULL, NULL, &root_fd);
+        if (r < 0)
+                return r;
+
+        r = fd_get_path(mntns_fd, &mntns);
+        if (r < 0)
+                return r;
+
+        r = namespace_open(getpid_cached(), NULL, &self_mntns_fd, NULL, NULL, NULL);
+        if (r < 0)
+                return r;
+
+        r = fd_get_path(self_mntns_fd, &self_mntns);
+        if (r < 0)
+                return r;
+
+        /* We can't add new mounts at runtime if the process wasn't started in a namespace */
+        if (streq(self_mntns, mntns)) {
+                if (error_message)
+                        *error_message = strdup("Failed to activate bind mount in target, not running in a mount namespace");
+                return -EINVAL;
+        }
```

Comment:
hmm, so this works, but I think it would be a lot nicer to compare by inode/dev_t. i.e. fstat() both fds, and compare `.st_ino and .st_dev`.

The /proc/fd/ symlinks are always a bit finicky i'd say, as the `(deleted)` stuff is weird, and procfs needs to be mounted, and string comparisons are strange too. the fstat() compare is much nicer I think.

Reply:
ok, done